### PR TITLE
Key/Value Question Bug Fix

### PR DESCRIPTION
**Pull request by GavinBurris42**
Issue #, if available:

Description of changes:  Fixed bug with Key/Value question when using backspace on text when no tags have been created yet.
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/ec2helper/ec2helper.go [hunk 1]
```diff
@@ -1493,6 +1491,7 @@ func (h *EC2Helper) LaunchFleet(templateId *string) (*ec2.CreateFleetOutput, err
 	}
 
 	result, err := h.Svc.CreateFleet(input)
+	fmt.Printf("Errors: %#v\n Length: %d", result.Errors, len(result.Instances))
```
Comment (snay2) [inline pkg/ec2helper/ec2helper.go:1494]: Do you want to keep this line?

File: pkg/questionModel/keyValue.go [hunk 2]
```diff
@@ -247,14 +247,16 @@ func (kv *KeyValue) cycleFocusIndex(msgType tea.KeyType) {
 }
 
 // deleteTag deletes a tag if cursor is focused on a tag in the tag list
```
Comment (snay2) [inline pkg/questionModel/keyValue.go:249]: Can you update this comment to describe what the boolean return value means?

File: pkg/ec2helper/ec2helper.go [hunk 3]
```diff
@@ -1475,7 +1473,7 @@ func (h *EC2Helper) LaunchFleet(templateId *string) (*ec2.CreateFleetOutput, err
 	}
 
 	spotRequest := &ec2.SpotOptionsRequest{
-		AllocationStrategy: aws.String("capacity-optimized"),
+		AllocationStrategy: aws.String("capacity-optimized-prioritized"),
```
Comment (snay2) [inline pkg/ec2helper/ec2helper.go:1476]: Why did this change? The line on the main repo is just `capacity-optimized`.

https://github.com/awslabs/aws-simple-ec2-cli/blob/main/pkg/ec2helper/ec2helper.go#L1456

File: cmd/launch.go [hunk 4]
```diff
@@ -225,6 +225,8 @@ func launchNonInteractive(h *ec2helper.EC2Helper) {
 		}
 	}
 
+	h.ChangeRegion(simpleConfig.Region)
```
Comment (snay2) [inline cmd/launch.go:228]: This is unrelated to the stated purpose of this PR. Are you sure it is correct and belongs?

File: pkg/ec2helper/ec2helper.go [hunk 5]
```diff
@@ -305,17 +305,15 @@ func (h *EC2Helper) GetInstanceTypesFromInstanceSelector(instanceSelector Instan
...
 
 	// Create filters for filtering instance types
 	vcpusRange := selector.IntRangeFilter{
 		LowerBound: vcpusLower,
-		UpperBound: vcpusUpper,
 	}
 	memoryRange := selector.ByteQuantityRangeFilter{
 		LowerBound: bytequantity.FromGiB(memoryLower),
-		UpperBound: bytequantity.FromGiB(memoryUpper),
 	}
```
Comment (snay2) [inline pkg/ec2helper/ec2helper.go:317]: Do these changes belong here? Are they correct?

File: pkg/ec2helper/ec2helper.go [hunk 6]
```diff
@@ -1116,7 +1114,7 @@ func (h *EC2Helper) LaunchSpotInstance(simpleConfig *config.SimpleInfo, detailed
 	if confirmation {
 		fmt.Println("Options confirmed! Launching spot instance...")
 		if simpleConfig.LaunchTemplateId != "" {
-			_, err = h.LaunchFleet(aws.String(simpleConfig.LaunchTemplateId))
+			_, err = h.LaunchFleet(&simpleConfig.LaunchTemplateId)
```
Comment (snay2) [inline pkg/ec2helper/ec2helper.go:1117]: Is this change related to the key/value question fix?

**Review by snay2 — CHANGES_REQUESTED** [review]
This PR has a lot of commits, and many of them seem unrelated to this change. Can you confirm that the diff is complete and correct? Otherwise, it might be better to open a PR with just the relevant commit.

File: pkg/ec2helper/ec2helper.go (hunk 1 above)
Comment (GavinBurris42) [reply pkg/ec2helper/ec2helper.go:1494]: Nope deleted

File: pkg/ec2helper/ec2helper.go (hunk 3 above)
Comment (GavinBurris42) [reply pkg/ec2helper/ec2helper.go:1476]: Reverted

File: cmd/launch.go (hunk 4 above)
Comment (GavinBurris42) [reply cmd/launch.go:228]: Necessary for bug that occurs when instance region isnt the same as local session region
